src/relations/peers.py

```diff
@@ -0,0 +1,255 @@
+            return
+
+        if cfg := self.app_databag.get(CFG_FILE_DATABAG_KEY):
+            self.charm.render_pgb_config(PgbConfig(cfg))
+
+        if auth_file := self.app_databag.get(AUTH_FILE_DATABAG_KEY):
+            self.charm.render_auth_file(auth_file)
+
+        if cfg is not None or auth_file is not None:
+            try:
+                # returns an error if this is fired before on_pebble_ready.
+                self.charm.reload_pgbouncer()
+            except ConnectionError:
+                # TODO find a better error to use
+                event.defer()
+
+    def set_secret(self, scope:str, key:str, value:str):
+        """Sets secret value.
+
+        Placeholder method for Juju Secrets interface.
+
+        Args:
+            scope: scope for data
+            key: key to set value to
+            value: value to be set
+        """
+        if scope == "unit":
+            if not value:
+                self.unit_databag.pop(key, None)
+                return
+            self.unit_databag.update({key: value})
+        elif scope == "app":
+            if not value:
+                self.app_databag.pop(key, None)
+                return
+            self.app_databag.update({key: value})
+        else:
+            raise RuntimeError("Unknown secret scope.")
+
+    def del_secret(self, scope:str, key:str):
+        """Deletes secret value.
+
+        Placeholder method for Juju Secrets interface.
+
+        Args:
+            scope: scope for data
+            key: key to access data
+        """
+        if scope == "unit":
+            self.unit_databag.pop(key, None)
+            return
+        elif scope == "app":
+            self.app_databag.pop(key, None)
+            return
+        else:
+            raise RuntimeError("Unknown secret scope.")
+
+    def get_secret(self, scope:str, key:str) -> Optional[str]:
+        """Gets secret value.
+
+        Placeholder method for Juju Secrets interface.
+
+        Args:
+            scope: scope for data
+            key: key to access data
+        Returns:
+            value at `key` in `scope` databag.
+        """
+        if scope == "unit":
+            return self.unit_databag.get(key, None)
+        elif scope == "app":
+            return self.app_databag.get(key, None)
+        else:
+            raise RuntimeError("Unknown secret scope.")
+
+    def update_cfg(self, cfg: PgbConfig) -> None:
+        """Writes cfg to app databag if leader."""
+        if not self.charm.unit.is_leader():
+            return
+
+        if self.app_databag is None:
+            # peer relation not yet initialised
+            # TODO fail louder
+            return
+
+        self.app_databag[CFG_FILE_DATABAG_KEY] = cfg.render()
+        logger.debug("updated config file in peer databag")
+
+    def get_cfg(self) -> PgbConfig:
+        """Retrieves the pgbouncer config from the peer databag."""
+        if cfg := self.get_secret("app", CFG_FILE_DATABAG_KEY):
```

Comment:
Is this supposed to be a secret? If so, you can also use `set_secret` on L225 and L142.